### ISSUE TITLE
Deleted messages from Pulse aren't actually deleted from phone's database

### ISSUE THREAD

**Issue by SpookDroid** [klinker-apps/messenger-issues#803]
- [x ] I have [searched](https://github.com/klinker-apps/messenger-issues/issues), by keyword, for my problem to verify that there aren't any open or closed issues around this behavior.
- [x ] I have [searched the help page](https://messenger.klinkerapps.com/help) to verify that it does not have any topics relevant to this issue.
- [ x] I have verified that this is something that is reproducible.

**Device with issue:** Galaxy Note 9, Android 8.1.0,
**Primary device:** Galaxy Note 9
**App version:** Beta version 4.3.3.2300

**Describe the bug:**
I recently noticed that any message/thread deleted in Pulse will not actually be deleted from the phone's database.  It is deleted IN Pulse (app, web version, and tablet app synced), but if I open the stock messaging app (or other 3rd party apps), the supposedly-deleted messages are still there (again, just not visible in Pulse). 

I tried changing defaults to the stock app, Textra, and Google's Messages and they all delete messages/threads properly, but when switching back to Pulse the same issue happens: any message I delete there will still be visible for all other apps.  

All messages deleted since switching to Pulse were still visible in the stock and other 3rd party apps, so not really something that started with this version. 

**Steps to reproduce:**

1. Delete any thread/message in Pulse
2. Open stock messaging app or other 3rd party ones
3. 'Deleted' messages are still in those apps, but not in Pulse

**Expected behavior:**
Deleted messages should also be deleted from phone's database. 

**Screenshots:**

**Additional information:**
I tried clearing permissions and re-enabling them, but nothing helped.  Pulse has all permissions active.

**Comment by SpookDroid** [comment]
Well, I just tried that and nope, deleting a thread yields the same issue 😔

Bummer.

**Comment by ta79pr** [comment]
Yes, sometimes I get a sensitive work text that should be deleted without deleting the whole conversation. Yet, deleted message appears in Android contacts and stock contacts app.

**Comment by SpookDroid** [comment]
As the developer mentioned that's expected behavior.   And I get it, it does suck, but it's also work to implement haha.  

Unfortunately for me it seems that not even threads are getting deleted, so that's a bit worse.  Still beats having to use other apps as Pulse rocks, so I just added a reminder to clean up my inbox (in my case, fortunately, all the messages/threads I need to delete have the same 'unique keyword' in them so it makes it a bit easy to purge).

**Comment by ta79pr** [comment]
Well, is there a way to actually disable the stock messages app? That would solve the problem right?

**Comment by klinker24** [comment]
You don't have to have the stock app installed or enabled. Pulse is standalone.

**Comment by SpookDroid** [comment]
Still wouldn't solve the problem though; the messages wouldn't be deleted
(just now you wouldn't have another app to view and actually delete them
from).

**Comment by klinker24** [comment]
He asked a questions, so I answered it. I agree that it does nothing for the request in this issue. Sorry, solving this is not a huge priority for me, at this time.

**Comment by SpookDroid** [comment]
Sorry didn't mean to sound rude or anything! Haha that was also for him, he
seemed to think disabling the stock app would solve this. I understand it's
in the bottom pile hehe. Can't be easy doing all this on your own, so
thanks for making this awesome app.

**Comment by ta79pr** [comment]
> You don't have to have the stock app installed or enabled. Pulse is standalone.

Understood. Then my issue is simply my inability to disable/uninstall the built-in Galaxy "Messages" app. In my ignorance, I thought the built-in app was required to send/receive messages. Thanks for the replies, I understand things better now.

**Comment by el-pyrolad** [comment]
> > You don't have to have the stock app installed or enabled. Pulse is standalone.
> 
> Understood. Then my issue is simply my inability to disable/uninstall the built-in Galaxy "Messages" app. In my ignorance, I thought the built-in app was required to send/receive messages. Thanks for the replies, I understand things better now.

There are methods that allow you to disable built-in apps without having to root your device or void your warranty. Also, these methods are reversible. I use a package disabler app to disable bloat from my AT&T Samsung Galaxy Note8 (Galaxy "Messages" app included). Pulse is not impacted at all -- it works great!

**Comment by weaverinva** [comment]
I am having this issue as well.  Individual messages including individual pictures are not deleted from the internal database.  I backup my full messages database often and move between phones.  Pictures take up space.  I have to go to another message app to delete the individual pictures within a conversation thread.  Please fix this soon.  Should be straightforward.  Thank you for considering.

**Comment by weaverinva** [comment]
Hello Mr. Klinker, Any update on the priority of this situation of individual messages and picture deletion?  Now that we are 18 months later?  Thank you, Mike

**Comment by klinker24** [comment]
If I have an update, I will post it. This issue is not currently a priority for me. If you feel as though it is straightforward, feel free to fix it yourself: https://github.com/klinker-apps/pulse-sms-android

If you are not willing to explore fixing this yourself, please do not try to make the assumption or assertion that something is straight-forward.

**Comment by weaverinva** [comment]
Thank you very much for your response, Mr. Klinker. I apologize for offending you and retract the use of the word 'straight forward'.  I have used many other SMS clients including OEM and third party, and in my limited experience of using 4-6 different clients, they all deleted individual messages from internal database. Your architecture and functionality are innovative; thank you for continuing to support this app.  Peace.

**Comment by DarkWolf7** [comment]
By the way, this also affects the auto deletion of old messages. They are not being cleaned up from the phones database so the sms database is just getting bigger and bigger, and thanks to Google the app I used to use to take care of old messages is no longer available.

**Comment by weaverinva** [comment]
@DarkWolF7 I have over 12000 text messages in my database and it is not a problem to backup  and restore all of them. The key is to delete the pictures. Every now and then I change messaging apps and re delete MMS pictures. Par for the course right now.

**Comment by DarkWolf7** [comment]
> 
> 
> @DarkWolf7 I have over 12000 text messages in my database and it is not a problem to backup and restore all of them. The key is to delete the pictures. Every now and then I change messaging apps and re delete MMS pictures. Par for the course right now.

I hadn't realized it until I noticed that my backups were sitting @ 2gb which is a huge chunk out of my 15gb google drive limit. When I looked I had messages all the way back to 2018, along with images. I started doing what you're doing, but I wanted to make sure that when (if?) it's fixed, the auto-delete functionality is also addressed.
